src/test/scala/org/ergoplatform/mining/ErgoMinerSpec.scala

```diff
@@ -60,7 +71,135 @@ class ErgoMinerSpec extends TestKit(ActorSystem()) with FlatSpecLike with ErgoTe
       blocksGenerated should be > 1
       blocksGenerated should be >= height
     }
+
+    system.terminate()
+  }
+
+  it should "filter out double spend txs" in {
+    val tx = validErgoTransactionGen.sample.get._2
+    ErgoMiner.fixTxsConflicts(Seq(tx, tx, tx)) should have length 1
+
+    val inputs = validErgoTransactionGenTemplate(0, -1, 100, 100).sample.get._1
+    val (l, r) = inputs.splitAt(50)
+    val tx_1 = validTransactionGen(l).sample.get
+    val tx_2 = validTransactionGen(r :+ l.last).sample.get
+
+    ErgoMiner.fixTxsConflicts(Seq(tx_1, tx_2, tx)) should contain theSameElementsAs Seq(tx_1, tx)
+    ErgoMiner.fixTxsConflicts(Seq(tx_2, tx_1, tx)) should contain theSameElementsAs Seq(tx_2, tx)
+  }
+
+  it should "create own coinbase transaction, if there is already a transaction, that spends emission box" in new TestKit(ActorSystem()){
```

Comment:
When I looks to Miner code, I can't understand, how it works. Can you describe, why this test passed?

Reply:
There is a mandatory that only miner can spent emission box, so every inputs of emission box outside of coinbase filtering out from block.